libs/net/dal/Services/FileReferenceService.cs

```diff
@@ -254,5 +258,155 @@ public async Task<IEnumerable<FileReference>> GetFiles(DateTime? updatedBefore =
             return Enumerable.Empty<FileReference>();
         }
     }
-    #endregion
+
+    public async Task<Stream> DownloadFromS3Async(string s3Key)
+    {
+        var accessKey = Environment.GetEnvironmentVariable("S3_ACCESS_KEY") ?? throw new InvalidOperationException("S3_ACCESS_KEY environment variable is not set");
```

Comment:
Move all of these into dependency injected configuration options.  Look into the `/api/net/Config` folder for examples